### PR TITLE
transaction: Rename is_coin_base to is_coinbase

### PR DESCRIPTION
**Pull request by stevenroose**
Alternative to https://github.com/rust-bitcoin/rust-bitcoin/pull/1795.

Keep the old method as deprecated and add doc alias. Also change internal usage of the method.

### CODE REVIEW

**Review by Kixunil — DISMISSED** [review]
ACK b2e90eaabfa7c7683882484ae5dfc137f787190f

I prefer this to the alternative.

**Comment by apoelstra** [comment]
Lolll looks like we need to stick `allow(clippy::deprecated_semver)` on the crate to use the `NEXT_VERSION` trick.

File: bitcoin/src/blockdata/transaction.rs [hunk 1]
```diff
@@ -981,10 +981,19 @@ impl Transaction {
...
     /// all-zeros (creates satoshis "out of thin air").
-    pub fn is_coin_base(&self) -> bool {
+    #[doc(alias = "is_coin_base")] // method previously had this name
+    pub fn is_coinbase(&self) -> bool {
         self.input.len() == 1 && self.input[0].previous_output.is_null()
     }
 
+    /// Checks if this is a coinbase transaction.
+    ///
+    /// This method is deprecated, use [is_coinbase] instead.
```
Comment (tcharding) [inline bitcoin/src/blockdata/transaction.rs:991]: We don't need these lines, the `deprecated` attribute adds all the doc markers we need, right?

File: bitcoin/src/blockdata/transaction.rs [hunk 2]
```diff
@@ -981,10 +981,19 @@ impl Transaction {
     /// transaction. It is impossible to check if the transaction is first in the block, so this
     /// function checks the structure of the transaction instead - the previous output must be
     /// all-zeros (creates satoshis "out of thin air").
-    pub fn is_coin_base(&self) -> bool {
+    #[doc(alias = "is_coin_base")] // method previously had this name
```
Comment (tcharding) [inline bitcoin/src/blockdata/transaction.rs:984]: Interesting, is the idea to remove this when we remove the deprecated function? I've never used this attribute myself.

**Review by tcharding — APPROVED** [review]
ACK 5ad3da256ba221f945be275ac2e6dba0398f6950

**Comment by tcharding** [comment]
Looks like "NEXT_RELEASE" will need to go in in the "note = NEXT RELEASE" field. We should grep for "NEXT.RELEASE" too so we don't miss "NEXT-RELEASE" or "NEXT RELEASE", and leave that in released code which would be pretty embarrassing.

**Comment by Kixunil** [comment]
We could also allow the lint in non-release PRs and enable it in release PRs.

**Comment by apoelstra** [comment]
> We could also allow the lint in non-release PRs and enable it in release PRs.

I like this. I think we do it by `cargo clippy -- -D clippy::deprecated_semver` ... though I tried to test this, and we don't have any globally-allowed lints I can test this on, and it seems like it doesn't override locally-allowed ones.

File: bitcoin/src/blockdata/transaction.rs (hunk 2 above)
Comment (stevenroose) [reply bitcoin/src/blockdata/transaction.rs:984]: I'd keep it for as long as we can tolerate it. Anyone might remember the old name and search for it. It doesn't hurt I guess, it only makes it show up in search when you search for that name.

File: bitcoin/src/blockdata/transaction.rs (hunk 1 above)
Comment (stevenroose) [reply bitcoin/src/blockdata/transaction.rs:991]: fixing

**Comment by stevenroose** [comment]
Grrrr at clippy enforcement, it's so unproductive! So I will try to silence clippy for non-release CI then? Is that the consensus? Let me see if I can figure out how to do that.

**Comment by stevenroose** [comment]
Hmm, I'm not sure how the CI is built, but running clippy in the release job would require having nightly cargo in that job and I don't think we want that?

I went for what Tobin suggested instead, renaming to `NEXT-RELEASE` and matching for that. That seems to be accepted by overlord clippy nightly as a valid semver version. Tbh I think I've seen underscores in semvers, I don't see the issue.  I tested the grep and for me it worked.

File: bitcoin/src/blockdata/transaction.rs (hunk 2 above)
Comment (Kixunil) [reply bitcoin/src/blockdata/transaction.rs:984]: Yeah, I wouldn't mind keeping it forever.

**Comment by Kixunil** [comment]
Why would nightly be a problem? Sure there's a risk that release gets stalled a bit due to a nightly bug but worst case we could turn it off.

**Comment by stevenroose** [comment]
> Why would nightly be a problem? Sure there's a risk that release gets stalled a bit due to a nightly bug but worst case we could turn it off.

Nightly is generally a problem, right? What if a regression in nightly actually makes the release go through while it shouldn't? In fact I'd say it makes most sense to test the release in MSRV. Idk exactly what the release command checks for and how it differs from `cargo build`, but I can imagine things like new manifest fields or something be a thing. We don't have `Cargo.lock`, so that should be ok.

Anyway, I think like it's done in my latest commit should make sense.

**Comment by Kixunil** [comment]
I believe we should test on all versions.

**Review by tcharding — APPROVED** [review]
ACK 38d11ce3da6a6574c4db797d7e59b2adb51675fb

**Comment by tcharding** [comment]
CI fail is a formatting issue in patch 1 bro. You're favourite :)

**Review by tcharding — APPROVED** [review]
ACK a54e1ceab1bd3ef61a45153725a545a5a35c8138

**Review by apoelstra — APPROVED** [review]
ACK a54e1ceab1bd3ef61a45153725a545a5a35c8138

**Comment by tcharding** [comment]
Lol, just noticed the author of the last commit.

**Pull request merged**